packages/trader/src/Modules/Trading/Containers/trade-params-mobile.tsx

```diff
@@ -224,7 +225,29 @@ const TradeParamsMobile = observer(
         h_duration,
         d_duration,
     }: TTradeParamsMobile) => {
-        const { basis_list, basis, expiry_epoch, is_turbos, is_vanilla } = useTraderStore();
+        const {
+            basis_list,
+            basis,
+            expiry_epoch,
+            is_turbos,
+            is_vanilla,
+            duration_unit: selected_duration_unit,
+        } = useTraderStore();
+
+        React.useEffect(() => {
+            const toggled_duration_tab_idx = duration_units_list.findIndex(d => d.value === duration_unit);
+            const selected_duration_tab_idx = duration_units_list.findIndex(d => d.value === selected_duration_unit);
+            const defaultDuration = makeGetDefaultDuration(0, 'default')(duration_unit);
+
+            if (has_duration_error) {
+                setSelectedDuration(duration_unit, defaultDuration);
+            } else {
+                setSelectedDuration(duration_unit, duration_value);
+            }
```

Comment:
```suggestion
            setSelectedDuration(duration_unit, has_duration_error ? defaultDuration : duration_value);
```

Reply:
yeah i tried that. Broke the logic so i reverted lol